### PR TITLE
F#16 web server

### LINKED ISSUE #16: Create web package
Create ROS package which will:

- start web server which can run nodejs 
- renders page with text "Hello World"

### PR DESCRIPTION
**Pull request by MaxZhaloba**
Almost closes #16.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,3 +103,26 @@ Save map to file
 ```bash
 rosrun map_server map_saver -f <map_file_name>
 ```
+
+## Starting the web server
+
+This module requires up-to-date system packages and additional rosdep and npm dependencies before running.
```
Comment (AndriyPt) [inline README.md]: Please remove reference to contributing.md. All developers should follow contributing document and not only web server update will require dependencies installation. Else we will end up having these instructions for all new launch modules :)

File: leobot_web_server/CMakeLists.txt [hunk 2]
```diff
@@ -2,3 +2,10 @@ cmake_minimum_required(VERSION 2.8.3)
 project(leobot_web_server)
 
 find_package(catkin REQUIRED)
+
```
Comment (AndriyPt) [inline leobot_web_server/CMakeLists.txt:5]: please check how minimalistic CMakeList.txt could look like https://github.com/AndriyPt/ros-navigation-examples/blob/kinetic-devel/moon_control/CMakeLists.txt

File: leobot_web_server/web_content/index.html [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Leobot</title>
+    <link rel="stylesheet" href="style.css">
+</head>
+<body>
+    <header>
+        <div class="header-wrapper">
+            <div class="header-image"></div>
+            <h1>OPEN SOURCE TELEPRESENSE ROBOT</h1>
```
Comment (AndriyPt) [inline leobot_web_server/web_content/index.html]: Please fix to *TELEPRESENCE*

File: leobot_web_server/web_content/index.html [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                <button class="small-button-top"></button>
+                <div class="small-left-right">
+                    <button class="small-button-left"></button>
+                    <button class="small-button-right"></button>
+                </div>
+                <button class="small-button-bottom"></button>
+        </div>
+    </main>
+    <footer>
+        <p>LEOBOT©2018 Created by SYSTEMDI <br>Developed by Yaryna Demkiv</p>
```
Comment (AndriyPt) [inline leobot_web_server/web_content/index.html]: please fix footnote according to https://github.com/ros-ukraine/leobot/issues/43

File: leobot_web_server/web_content/style.css [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+    justify-content: space-between;
+    margin-left: auto;
+    margin-right: auto;
+}
+
+.header-image {
+    width: 242px;
+    height: 52px;
+    margin-left: 15px;
+    background-image: url("http://res.cloudinary.com/dvhn8lvvz/image/upload/v1515331815/site-logo_xrpypd.png");
```
Comment (AndriyPt) [inline leobot_web_server/web_content/style.css]: please download image to images folder and refer it from there

File: leobot_web_server/web_content/style.css [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+    .header-wrapper {
+        width: 320px;
+        justify-content: center;
+    }
+
+    .header-image {
+        width: 43px;
+        height: 43px;
+        margin-left: 5px;
+        background-image: url(http://res.cloudinary.com/dvhn8lvvz/image/upload/c_fill,g_west,h_53,w_53/v1515331815/site-logo_xrpypd.png);
```
Comment (AndriyPt) [inline leobot_web_server/web_content/style.css]: please download image to *images* folder and refer it from there

File: leobot_web_server/web_content/style.css [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+
+    .control-panel-header {
+        font-size: 12px;
+    }
+    .small-button-top {
+        border: none;
+        background-color: transparent;
+        width: 30px;
+        height: 30px;
+        background-image: url("http://res.cloudinary.com/dvhn8lvvz/image/upload/c_scale,h_30,w_30/a_270/v1515333945/arrow-right-01-256_b40xbs.png");
```
Comment (AndriyPt) [inline leobot_web_server/web_content/style.css]: please download image to *images* folder and refer it from there

File: leobot_web_server/web_content/style.css [hunk 8]
```diff
@@ -0,0 +1,504 @@
...
+        display: flex;
+    }
+
+    .small-button-left {
+        border: none;
+        background-color: transparent;
+        width: 30px;
+        height: 30px;
+        margin-right: 40px;
+        background-image: url("http://res.cloudinary.com/dvhn8lvvz/image/upload/c_scale,h_30,w_30/a_180/v1515333945/arrow-right-01-256_b40xbs.png");
```
Comment (AndriyPt) [inline leobot_web_server/web_content/style.css]: please download image to *images* folder and refer it from there

File: leobot_web_server/web_content/style.css [hunk 9]
```diff
@@ -0,0 +1,504 @@
...
+    }
+
+    .small-button-right {
+        border: none;
+        background-color: transparent;
+        width: 30px;
+        height: 30px;
+        margin-left: 40px;
+        background-image: url("http://res.cloudinary.com/dvhn8lvvz/image/upload/c_scale,h_30,w_30/a_0/v1515333945/arrow-right-01-256_b40xbs.png");
```
Comment (AndriyPt) [inline leobot_web_server/web_content/style.css]: please download image to *images* folder and refer it from there

File: leobot_web_server/web_content/style.css [hunk 10]
```diff
@@ -0,0 +1,504 @@
...
+    }
+
+    .small-button-bottom {
+        border: none;
+        background-color: transparent;
+        width: 30px;
+        height: 30px;
+        background-image: url("http://res.cloudinary.com/dvhn8lvvz/image/upload/c_scale,h_30,w_30/a_90/v1515333945/arrow-right-01-256_b40xbs.png");
```
Comment (AndriyPt) [inline leobot_web_server/web_content/style.css]: please download image to *images* folder and refer it from there

File: leobot_web_server/scripts/web_server.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    console.log('rosnodejs handle: ' + nh);
+
+    nh.getParam('port').then((p) => {
+        const port = p;
+
+        console.log('Port parameter: ' + port);
+    });
+});
+
+const port = 8080; // hard-coded for now
```
Comment (AndriyPt) [inline leobot_web_server/scripts/web_server.js]: Please note that all code is located in initNode.then callback https://github.com/RethinkRobotics-opensource/rosnodejs_examples/blob/kinetic-devel/scripts/listener.js#L32-L41

File: leobot_web_server/scripts/web_server.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env nodejs
```
Comment (AndriyPt) [inline leobot_web_server/scripts/web_server.js:1]: please use this approach https://github.com/RethinkRobotics-opensource/rosnodejs_examples/blob/kinetic-devel/scripts/listener.js#L43

File: leobot_web_server/scripts/web_server.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+const fs = require('fs');
+const path = require('path');
+const rosnodejs = require('rosnodejs');
+
+rosnodejs.initNode('leobot_web_server').then(() => {
+    const nh = rosnodejs.nh;
+
+    console.log('rosnodejs handle: ' + nh);
+
+    nh.getParam('port').then((p) => {
```
Comment (AndriyPt) [inline leobot_web_server/scripts/web_server.js]: please check this example how parameters are read https://github.com/RethinkRobotics-opensource/rosnodejs/blob/kinetic-devel/src/examples/standard.js#L44-L46

**Review by AndriyPt — CHANGES_REQUESTED** [review]
Few remarks

**Comment by MaxZhaloba** [comment]
I've fixed the simple parts of functionality, but it appears that I don't understand some other ones.

Can we please have a chat on the following items:
1. catkin_package instruction in CMakeLists.txt
2. Conditional listener and initNode in JS file
3. Reading of parameters using rosNode.getParam

**Comment by MaxZhaloba** [comment]
@AndriyPt, I've fixed all current issues. Feel free to review.

**Comment by AndriyPt** [comment]
@MaxZhaloba Travis CI is failing. Please make sure that all CI servers are green before requesting re-reviewing of the code :)
The error is
```
CMake Error at cmake_install.cmake:133 (file):
  file INSTALL cannot find
  "/home/user/workspace/src/host/home/travis/build/ros-ukraine/leobot/leobot_web_server/scripts/**/*".
```

**Comment by MaxZhaloba** [comment]
Fixed

**Review by AndriyPt — APPROVED** [review]

**Pull request merged**